test/test_selectors.py

```diff
@@ -119,86 +118,15 @@ def assertTakesTime(self, lower=None, upper=None):
-class WaitForIOTest(unittest.TestCase, AlarmMixin, TimerMixin):
-    """ Tests for the higher level wait_for_* functions. """
-
-    def make_socketpair(self):
-        rd, wr = socketpair()
-
-        rd.settimeout(0.0)
-        wr.settimeout(0.0)
-
-        self.addCleanup(rd.close)
-        self.addCleanup(wr.close)
-        return rd, wr
-
-    def test_selector_error(self):
-        err = selectors.SelectorError(1)
-        self.assertEqual(err.__repr__(), "<SelectorError errno=1>")
-        self.assertEqual(err.__str__(), "<SelectorError errno=1>")
-
-    def test_wait_for_read_single_socket(self):
-        rd, wr = self.make_socketpair()
-        self.assertEqual([], wait_for_read(rd, timeout=SHORT_SELECT))
-
-    def test_wait_for_read_multiple_socket(self):
-        rd, rd2 = self.make_socketpair()
-        self.assertEqual([], wait_for_read([rd, rd2], timeout=SHORT_SELECT))
-
-    def test_wait_for_read_empty(self):
-        self.assertEqual([], wait_for_read([], timeout=SHORT_SELECT))
-
-    def test_wait_for_write_single_socket(self):
-        wr, wr2 = self.make_socketpair()
-        self.assertEqual([wr], wait_for_write(wr, timeout=SHORT_SELECT))
-
-    def test_wait_for_write_multiple_socket(self):
-        wr, wr2 = self.make_socketpair()
-        result = wait_for_write([wr, wr2], timeout=SHORT_SELECT)
-        # assertItemsEqual renamed in Python 3.x
-        if hasattr(self, "assertItemsEqual"):
-            self.assertItemsEqual([wr, wr2], result)
-        else:
-            self.assertCountEqual([wr, wr2], result)
-
-    def test_wait_for_write_empty(self):
-        self.assertEqual([], wait_for_write([], timeout=SHORT_SELECT))
-
-    def test_wait_for_non_list_iterable(self):
-        rd, wr = self.make_socketpair()
-        iterable = {'rd': rd}.values()
-        self.assertEqual([], wait_for_read(iterable, timeout=SHORT_SELECT))
-
-    def test_wait_timeout(self):
-        rd, wr = self.make_socketpair()
-        with self.assertTakesTime(lower=SHORT_SELECT, upper=SHORT_SELECT):
-            wait_for_read([rd], timeout=SHORT_SELECT)
-
-    @skipUnless(HAS_ALARM, "Platform doesn't have signal.alarm()")
-    def test_interrupt_wait_for_read_no_event(self):
-        rd, wr = self.make_socketpair()
-
-        self.set_alarm(SHORT_SELECT, lambda *args: None)
-
-        with self.assertTakesTime(lower=LONG_SELECT, upper=LONG_SELECT):
-            self.assertEqual([], wait_for_read(rd, timeout=LONG_SELECT))
-
-    @skipUnless(HAS_ALARM, "Platform doesn't have signal.alarm()")
-    def test_interrupt_wait_for_read_with_event(self):
-        rd, wr = self.make_socketpair()
-
-        self.set_alarm(SHORT_SELECT, lambda *args: wr.send(b'x'))
-
-        with self.assertTakesTime(lower=SHORT_SELECT, upper=SHORT_SELECT):
-            self.assertEqual([rd], wait_for_read(rd, timeout=LONG_SELECT))
-        self.assertEqual(rd.recv(1), b'x')
-
-
-@skipUnless(selectors.HAS_SELECT, "Platform doesn't have a selector")
```

Comment:
So I understand why you moved all this into the `BaseSelectorTestCase`, but I think in this instance I'd rather have some copy-paste code to get the "runs with every selector" behaviour. I quite liked the logical separation of the tests before.